torchmetrics/detection/map.py

```diff
@@ -664,19 +663,17 @@ def __calculate_recall_precision_scores(
             recall[idx, idx_cls, idx_bbox_area, idx_max_det_thrs] = rc[-1] if nd else 0
 
             # Remove zigzags for AUC
-            for i in range(nd - 1, 0, -1):
-                if pr[i] > pr[i - 1]:
-                    pr[i - 1] = pr[i]
+            diff_zero = torch.zeros((1,))
+            diff = torch.ones((1,))
+            while not torch.all(diff == 0):
+                diff = torch.clamp(torch.cat((torch.diff(pr), diff_zero), 0), min=0)
```

Comment:
`torch.diff` was first introduced in v1.8. This solution is better for backwards compatibility.
```suggestion
                diff = torch.clamp(torch.cat((pr[1:] - pr[:-1], diff_zero), 0), min=0)
```